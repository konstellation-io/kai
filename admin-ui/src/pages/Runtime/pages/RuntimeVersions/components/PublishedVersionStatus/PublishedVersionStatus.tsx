import React from 'react';
import ROUTE from '../../../../../../constants/routes';
import { useParams } from 'react-router';

import Button from '../../../../../../components/Button/Button';
import { formatDate } from '../../../../../../utils/format';

import Check from '@material-ui/icons/Check';
import ErrorOutline from '@material-ui/icons/ErrorOutline';

import { Version } from '../../../../../../graphql/models';

import cx from 'classnames';
import styles from '../../RuntimeVersions.module.scss';

type Props = {
  publishedVersion?: Version;
};
function PublishedVersionStatus({ publishedVersion }: Props) {
  const { runtimeId } = useParams();
  const isVersionActive = publishedVersion !== undefined;
  const title = isVersionActive
    ? 'Version published'
    : 'There is no published version';
  const Icon = isVersionActive ? Check : ErrorOutline;

  const newVersionRoute = ROUTE.NEW_VERSION.replace(
    ':runtimeId',
    runtimeId || ''
  );

  return (
    <div
      className={cx(styles.activeVersion, {
        [styles['active']]: isVersionActive
      })}
    >
      <Icon className="icon-regular" />
      <span className={styles.versionTitle}>{title}</span>
      {isVersionActive && publishedVersion && (
        <>
          <span className={styles.versionCreation}>
            {`CREATED: ${formatDate(new Date(publishedVersion.creationDate))}`}
          </span>
          <div className={styles.secondRow}>
            <div className={styles.versionStatus}>
              <div className={styles.greenCircle} />
              <span className={styles.versionName}>
                {publishedVersion.name}
              </span>
            </div>
            <Button label="ADD NEW VERSION" to={newVersionRoute} border />
          </div>
        </>
      )}
    </div>
  );
}

export default PublishedVersionStatus;
