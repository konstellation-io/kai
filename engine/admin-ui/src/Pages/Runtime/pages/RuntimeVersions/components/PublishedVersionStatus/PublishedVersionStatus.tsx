import { Button } from 'kwc';
import Can from 'Components/Can/Can';
import ROUTE from 'Constants/routes';
import React from 'react';
import cx from 'classnames';
import styles from '../../RuntimeVersions.module.scss';

type Props = {
  nPublishedVersions: number;
  noVersions: boolean;
};
function PublishedVersionStatus({ noVersions, nPublishedVersions }: Props) {
  let title;
  if (noVersions) {
    title =
      'There are no runtime versions. Please, upload a new version to start working on this runtime.';
  } else if (nPublishedVersions >= 1) {
    title = `${nPublishedVersions} version${
      nPublishedVersions > 1 ? 's' : ''
    } published`;
  } else {
    title = 'There is no published version';
  }

  return (
    <div
      className={cx(styles.activeVersion, {
        [styles['active']]: nPublishedVersions
      })}
    >
      <span className={styles.versionTitle}>{title}</span>
      <Can perform="version:edit">
        <Button
          label="ADD VERSION"
          to={ROUTE.NEW_VERSION}
          primary
          height={30}
          style={{ borderRadius: 2 }}
        />
      </Can>
    </div>
  );
}

export default PublishedVersionStatus;
