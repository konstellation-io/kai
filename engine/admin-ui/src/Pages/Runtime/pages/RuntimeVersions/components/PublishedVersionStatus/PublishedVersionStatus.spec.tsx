import { BrowserRouter } from 'react-router-dom';
import PublishedVersionStatus from './PublishedVersionStatus';
import React from 'react';
import { mount } from 'enzyme';

describe('PublishedVersionStatus', () => {
  let wrapper;

  beforeEach(() => {
    wrapper = mount(
      <BrowserRouter>
        <PublishedVersionStatus nPublishedVersions={5} noVersions={false} />
      </BrowserRouter>
    );
  });

  it('matches snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });
});
