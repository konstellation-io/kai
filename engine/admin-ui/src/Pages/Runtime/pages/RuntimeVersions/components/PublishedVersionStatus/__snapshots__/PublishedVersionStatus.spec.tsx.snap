// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PublishedVersionStatus matches snapshot 1`] = `
<BrowserRouter>
  <Router
    history={
      Object {
        "action": "POP",
        "block": [Function],
        "createHref": [Function],
        "go": [Function],
        "goBack": [Function],
        "goForward": [Function],
        "length": 1,
        "listen": [Function],
        "location": Object {
          "hash": "",
          "pathname": "/",
          "search": "",
          "state": undefined,
        },
        "push": [Function],
        "replace": [Function],
      }
    }
  >
    <PublishedVersionStatus
      nPublishedVersions={5}
      noVersions={false}
    >
      <div
        className="activeVersion active"
      >
        <span
          className="versionTitle"
        >
          5 versions published
        </span>
        <Can
          perform="version:edit"
        >
          <P
            height={30}
            label="ADD VERSION"
            primary={true}
            style={
              Object {
                "borderRadius": 2,
              }
            }
            to="/new_version"
          >
            <Link
              data-testid="buttonLink"
              to="/new_version"
            >
              <LinkAnchor
                data-testid="buttonLink"
                href="/new_version"
                navigate={[Function]}
              >
                <a
                  data-testid="buttonLink"
                  href="/new_version"
                  onClick={[Function]}
                >
                  <div
                    className="_1lqJX _3kQFd _pg7GC undefined"
                    onClick={[Function]}
                    onKeyPress={[Function]}
                    style={
                      Object {
                        "borderRadius": 2,
                        "height": 30,
                        "lineHeight": "30px",
                      }
                    }
                    tabIndex={-1}
                    title=""
                  >
                    <span>
                      ADD VERSION
                    </span>
                  </div>
                </a>
              </LinkAnchor>
            </Link>
          </P>
        </Can>
      </div>
    </PublishedVersionStatus>
  </Router>
</BrowserRouter>
`;
